### PR TITLE
refactor(storage): refactor hummock timer loop

### PR DESCRIPTION
**Pull request by Li0k**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

### <samp>🤖 Generated by Copilot at 91b0d0d</samp>

This pull request refactors and simplifies the timer and compaction logic in the `hummock` module of the `meta` crate. It removes unused or redundant code, introduces a new function `hummock_timer_task` that handles multiple timer events, and moves some functionality between the `compaction_scheduler.rs` and `manager/mod.rs` files. It also updates the `rpc/server.rs` file and the `manager/tests.rs` file accordingly.

## Checklist For Contributors

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Review by wcy-fdu — COMMENTED** [review]
Generally LGTM!

**Review by Little-Wallace — APPROVED** [review]
LGTM

File: src/meta/src/hummock/manager/mod.rs [hunk 1]
```diff
@@ -2144,47 +2106,168 @@ where
...
+
+            let mut stat_report_interval =
+                tokio::time::interval(std::time::Duration::from_secs(STAT_REPORT_PERIOD_SEC));
+            stat_report_interval.set_missed_tick_behavior(tokio::time::MissedTickBehavior::Delay);
+            stat_report_interval.reset();
+            let stat_report_trigger =
+                IntervalStream::new(stat_report_interval).map(|_| HummockTimerEvent::Report);
+
+            let mut compaction_heartbeat_interval =
+                tokio::time::interval(std::time::Duration::from_millis(1000));
```
Comment (hzxa21) [inline src/meta/src/hummock/manager/mod.rs:2141]: nits: let's use a const for the interval just like other intervals

File: src/meta/src/hummock/manager/mod.rs [hunk 2]
```diff
@@ -2144,47 +2106,168 @@ where
...
-                    _ = min_interval.tick() => {
-                    },
-                    // Shutdown
-                    _ = &mut shutdown_rx => {
-                        tracing::info!("Lsm stat reporter is stopped");
-                        return;
-                    }
-                }
+            const CHECK_PENDING_TASK_PERIOD_SEC: u64 = 300;
+            const STAT_REPORT_PERIOD_SEC: u64 = 10;
```
Comment (hzxa21) [inline src/meta/src/hummock/manager/mod.rs:2115]: nits: how about adding a log to print out all the timer interval on when starting the loop?

File: src/meta/src/hummock/manager/mod.rs [hunk 3]
```diff
@@ -2144,47 +2106,168 @@ where
...
+                        }
                     }
-                }
 
-                {
-                    hummock_manager.report_scale_compactor_info().await;
+                    Either::Right((_, _shutdown)) => {
+                        break;
+                    }
                 }
```
Comment (hzxa21) [inline src/meta/src/hummock/manager/mod.rs:2271]: nits: how about adding a log line when the loop exits?

**Pull request merged**